src/async_resolver.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{Prefix, ResolverError, Webfinger};
+use async_trait::async_trait;
+
+/// A trait to easily generate a WebFinger endpoint for any resource repository.
+///
+/// The `R` type is your resource repository (a database for instance) that will be passed to the
+/// [`find`](Resolver::find) and [`endpoint`](Resolver::endpoint) functions.
+#[async_trait]
+pub trait AsyncResolver {
+    type Repo: Send;
+    /// Returns the domain name of the current instance.
+    async fn instance_domain<'a>(&self) -> &'a str;
```

Comment:
I'm not sure this requires to be async, I would expect an implementation to return one of its fields, but I guess there is nothing wrong with too much async, it'll just get optimized away during monomorphisation

Reply:
and if it's `async` in the trait signature, it means you can use `await` in the function body.